### DISCUSSION TITLE
Allow users to add data to an existing table by csv

### DISCUSSION THREAD

**Discussion by fergusmeiklejohn** [supabase/supabase#2063] [Questions] [answered]
Our app needs to enable users to add data by uploading a CSV. Is this possible now, and/or will it be possible when we have functions?

Thanks!

**Comment by inian** [comment, marked as answer]
Hi @fergusmeiklejohn, you can parse the CSV on the client side and use our REST APIs to insert it into your table. Make sure that the table is locked down with Row level security and the users have insert access into the table. This is what our dashboard does too.

**Reply by fergusmeiklejohn** [reply]
Thanks Inian, yeah I was thinking about it overnight and I'd want to validate all the data and while users might have one csv I'd probably want to distribute that data over a few connected tables, so doing work client side and maybe in a cloud function is the way to do it..

**Reply by inian** [reply]
Yeah if you dont trust the client, doing it in a function may be the way to go.
